src/prometheus_server.py

```diff
@@ -14,15 +14,26 @@
 class Prometheus:
     """A class that represents a running instance of Prometheus."""
 
-    def __init__(self, host: str = "localhost", port: int = 9090, api_timeout=2.0):
+    def __init__(
+        self,
+        host: str = "localhost",
+        port: int = 9090,
+        web_route_prefix: str = "",
+        api_timeout=2.0,
+    ):
         """Utility to manage a Prometheus application.
 
         Args:
             host: Optional; host address of Prometheus application.
             port: Optional; port on which Prometheus service is exposed.
+            web_route_prefix: Optional; the root path added to the Prometheus API path, e.g.,
+              when we relate to an ingress.
             api_timeout: Optional; timeout (in seconds) to observe when interacting with the API.
         """
-        self.base_url = f"http://{host}:{port}"
+        if web_route_prefix and not web_route_prefix.startswith("/"):
+            web_route_prefix = f"/{web_route_prefix}"
+
+        self.base_url = f"http://{host}:{port}{web_route_prefix or ''}"
```

Comment:
Maybe use `urljoin` here? It will be both more clear and less error prone

```
>>> from urllib.parse import urljoin
>>> urljoin("http://localhost:9090", "/sub1")
'http://localhost:9090/sub1'
>>> urljoin("http://localhost:9090", "sub2")
'http://localhost:9090/sub2'
>>> urljoin("http://localhost:9090/", "sub3")
'http://localhost:9090/sub3'
>>> urljoin("http://localhost:9090/", "/sub4")
'http://localhost:9090/sub4'
```

Reply:
`urljoin` is now effectively my new favorite Python3 API.

Reply:
I take it back: `urljoin` does not work appending subpaths :-(

Reply:
So, in the end it is about ensuring the base URL ends with `/` and the path does not start with `/`. That was a LOT of footguns.